src/game_io/game_preload_packet.cc

```diff
@@ -111,7 +111,9 @@ void GamePreloadPacket::write(FileSystem& fs, Game& game, MapObjectSaver* const)
 	s.set_string("background", map.get_background());
 	s.set_string("win_condition", game.get_win_condition_displayname());
 	s.set_int("savetimestamp", static_cast<uint32_t>(time(nullptr)));
-	s.set_int("gametype", static_cast<int32_t>(game.game_controller()->get_game_type()));
+	s.set_int("gametype", static_cast<int32_t>(game.game_controller() != nullptr ?
+	                                              game.game_controller()->get_game_type() :
+	                                              GameController::GameType::kReplay));
```

Comment:
Please add a comment explaining why this should default to replay

Reply:
Because it needs to default to something and I couldn't think of anything better. Replay is the most general option.